### DISCUSSION TITLE
Render components of multiple containers

### DISCUSSION THREAD

**Discussion by artinnok** [structurizr/java#242] [Q&A]
### Description

Hi @simonbrowndotje!

First of all, I wanna say "thank you" for amazing tool set.

Secondly, I have a feature request that came to me from developers team.
For example, I have a following DSL:
```
workspace "C4" "Description" {

    model {
        softwareSystem C4 {
            container A {
                a1 = component A1
                a2 = component A2
            }
            container B {
                b1 = component B1
                b2 = component B2
            }
            
            a1 -> b1 "Makes API calls"
        }
    }

    configuration {
        scope softwaresystem
    }
}
```
Render of component level:

<img width="600" alt="source" src="https://github.com/structurizr/java/assets/8825727/7e06c287-fa87-47af-b83a-f3c2e3004ee0">

Developers are asked me to visualize **which component** of container B used by component A1. As I understand and reviewed the docs of Structurizr there is a limitation - render only components of **specific** container and it looks like it was based on whole approach of C4. 

But developers are really interested in details and expect to receive a detailed diagram in this particular case. I tried to provide example:

<img width="600" alt="c4" src="https://github.com/structurizr/java/assets/8825727/fe16efb0-fd2c-48ae-b404-58e2bec97498">

I hope that explanation of feature request was clear enough - feel free to reach me out with questions.

It would be neat to provide such feature.

Thanks,
Artem

### Priority

I have no budget and there's no rush, please add this feature for free

**Comment by simonbrowndotje** [comment]
> First of all, I wanna say "thank you" for amazing tool set.

Thanks very much - glad you're finding it useful!

> It would be neat to provide such feature.

You can do this already; for example:

```
workspace "C4" "Description" {

    model {
        softwareSystem C4 {
            a = container A {
                a1 = component A1
                a2 = component A2
            }
            b = container B {
                b1 = component B1
                b2 = component B2
            }
            
            a1 -> b1 "Makes API calls"
        }
    }
    
    views {
        component a "Containers-A" {
            include a1 a2 b1 b2
        }
    }

    configuration {
        scope softwaresystem
    }
}
```

![image](https://github.com/structurizr/java/assets/1009874/1696ba10-ad7b-41d5-9ee7-f70f0cccafa9)

See https://docs.structurizr.com/dsl/cookbook/container-view-multiple-software-systems/ for more (this is about containers, but the same approach applies).

**Reply by artinnok** [reply]
Thank you @simonbrowndotje!

Yep it works 👍
